kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigurationClient.java

```diff
@@ -139,13 +113,43 @@ static void removePropertySourceFromCache(String name) {
         return propertySources;
     }
 
+    /**
+     * Retrieves all of the {@link PropertySource} registrations for the given environment.
+     *
+     * @param environment The environment
+     * @return A {@link Publisher} that emits zero or many {@link PropertySource} instances discovered for the given environment
+     */
+    @Override
+    public Publisher<PropertySource> getPropertySources(Environment environment) {
+        if (!propertySources.isEmpty()) {
+            LOG.trace("Found cached PropertySources. Returning them");
+            return Flowable.fromIterable(propertySources.values());
+        } else {
+            LOG.trace("PropertySource cache is empty");
+            return getPropertySourcesFromConfigMaps().mergeWith(getPropertySourcesFromSecrets());
+        }
+    }
+
+    /**
+     * A description that describes this object.
+     *
+     * @return The description
+     */
+    @Override
+    public String getDescription() {
+        return KubernetesClient.SERVICE_ID;
+    }
+
     private Flowable<PropertySource> getPropertySourcesFromConfigMaps() {
         Predicate<KubernetesObject> includesFilter = KubernetesUtils.getIncludesFilter(configuration.getConfigMaps().getIncludes());
         Predicate<KubernetesObject> excludesFilter = KubernetesUtils.getExcludesFilter(configuration.getConfigMaps().getExcludes());
         Map<String, String> labels = configuration.getConfigMaps().getLabels();
-        String labelSelector = KubernetesUtils.computeLabelSelector(labels);
 
-        return Flowable.fromPublisher(client.listConfigMaps(configuration.getNamespace(), labelSelector))
+        return computePodLabels(client, configuration.getConfigMaps().getPodLabels(), configuration.getNamespace()).flatMap(podLabels -> {
```

Comment:
```
return computePodLabels(client, configuration.getConfigMaps().getPodLabels(), configuration.getNamespace()).flatMap(podLabels -> {
            podLabels.putAll(labels);
            String labelSelector = KubernetesUtils.computeLabelSelector(podLabels);

```

this code is duplicated there twice, generating the label selector. Encapsulating this into standalone method could improve the code readibility.